packages/blocks/src/api/factory.js

```diff
@@ -32,6 +32,7 @@ import {
 } from './registration';
 import {
 	normalizeBlockType,
+	__experimentalStripInternalBlockAttributes,
 	__experimentalSanitizeBlockAttributes,
```

Comment:
@kevin940726 @noisysocks It'd be good to look into whether this PR makes it possible to remove `__experimentalSanitizeBlockAttributes`, and update the widget editor to use an `internal` attribute instead.

Reply:
Maybe! `__internalWidgetId` can be set on **any** block type, so we'd have to hook into `registerBlockType` and add `widgetId: { type: 'string', __experimentalRole: 'internal' }` to the schema of every single registered block.

If we can do that then that will make `__experimentalCloneSanitizedBlock` unnecessary as this was only added so that duplicating blocks would not copy any attributes that don't exist in the schema e.g. `__internalWidgetId`. See https://github.com/WordPress/gutenberg/pull/29111.

It might be nice to try and do all of that along with this change so that there is a usage of `__experimentalRole: 'internal'` within core. We're less likely to break APIs that we ourselves use 🙂

Reply:
When discussing it in #29693 I envisioned that the API proposed in this PR would be sufficient to refactor the `__internalWidgetId` use case. It'd be great to confirm that.

Reply:
I can try implementing what I described above tomorrow unless @stacimc wants to give it a try 🙂

Reply:
Unfortunately I think this may require a little more tinkering -- as is, I'm running into the same issue in #29111 where internal attributes get dropped when the user makes changes in the widget editor 🤔 

In fact at the moment I'm seeing the issue even if I keep the separated `cloneBlock` and `__experimentalCloneSanitizedBlock`, because in order to fix the issue mentioned by @glendaviesnz where internal attributes aren't reset when you `Copy` + paste a block, I need to strip the internal attributes within `createBlock` as well. My somewhat tenuous understanding of the widget editor is that it fetches widgets and transforms them into blocks using `createBlock` -- so the internal attributes get reset every time you refresh the page. This ends up causing block invalidation errors.